### PR TITLE
Use core signature controller

### PR DESCRIPTION
**Pull request by matthewwalsh0**
**Description**

Replace direct usage of the message managers with the `SignatureController` provided by `@metamask/signature-controller` to improve consistency with the extension and to simplify the related code. 

**Checklist**

* [ ] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: temp/signature-controller/dist/SignatureController.js
Comment (matthewwalsh0) [inline temp/signature-controller/dist/SignatureController.js:1]: The dependency is loaded via this directory temporarily until `@metamask/signature-controller` is released.

**Comment by matthewwalsh0** [comment]
@SocketSecurity ignore-all

File: ios/Podfile.lock [hunk 1]
```diff
@@ -874,4 +874,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 74e503a275b4ed097cb01c171027882e9cdc699d
 
-COCOAPODS: 1.11.2
+COCOAPODS: 1.11.3
```
Comment (vinistevam) [inline ios/Podfile.lock:877]: Just out of curiosity, do you know what this file is about?

**Review by vinistevam — APPROVED** [review]

File: ios/Podfile.lock (hunk 1 above)
Comment (OGPoyraz) [reply ios/Podfile.lock:877]: It's a lock file for pods, like yarn.lock
I think this is getting updated with your cocoapods version, so we better leave it `1.11.2`

File: app/core/Engine.ts [hunk 2]
```diff
@@ -410,6 +406,34 @@ class Engine {
...
+              `${approvalController.name}:addRequest`,
+              `${approvalController.name}:acceptRequest`,
+              `${approvalController.name}:rejectRequest`,
+            ],
+          }),
+          isEthSignEnabled: () =>
+            Boolean(
+              preferencesController.state?.disabledRpcMethodPreferences
+                ?.eth_sign,
+            ),
```
Comment (bergarces) [inline app/core/Engine.ts:422]: I'm not even sure if regular signature is supported at all in mobile, but this should set it to false anyway.

**Review by bergarces — APPROVED** [review]

File: app/core/Engine.ts (hunk 2 above)
Comment (matthewwalsh0) [reply app/core/Engine.ts:422]: It's the same as the extension so there is a toggle in the advanced settings to enable it, have tested manually via the test dApp.

**Pull request merged**